File: array_pop/array_pop.php
<?php
// ATOMIC_FUNCTION: array_pop($a)
// TAGS: array
// LANG:  php
// FUNCTION: array_pop($a)

$start_val_1 = array("orange", "banana", "apple", "raspberry");
$expected_end_val = array("orange", "banana", "apple");
array_pop($start_val_1);
$end_val = $start_val_1;

// TEST
if($end_val == $expected_end_val) {
	echo "PASS";
} else {
	echo "FAIL";
}

?>

File: array_pop/array_pop.py

# ATOMIC_FUNCTION: array_pop(array)
# TAGS: list
# LANG:  python
# FUNCTION: array_pop(array)
def array_pop(array):
    n=len(array)
    if len(array)>1:
        return array[0:n-1]
    else:
        return None

start_val_1 = ["orange", "banana", "apple", "raspberry"]
expected_end_val = ["orange", "banana", "apple"]
end_val = array_pop(start_val_1)

# TEST
if(end_val == expected_end_val):
    print("PASS")
else:
    print("FAIL")